### PR TITLE
Support arrow keys for movement

### PR DESCRIPTION
**Pull request by benfrankel**

### CODE REVIEW

File: src/game/movement.rs [hunk 1]
```diff
@@ -13,32 +13,33 @@ pub(super) fn plugin(app: &mut App) {
...
+    // Need to normalize and scale because otherwise
+    // diagonal movement would be faster than horizontal or vertical movement.
+    let intent = intent.normalize_or_zero() * SPEED;
 
-        // Need to normalize and scale because otherwise
-        // diagonal movement would be faster than horizontal or vertical movement.
-        velocity.0 = velocity.normalize_or_zero() * SPEED;
+    for mut velocity in &mut velocity_query {
+        velocity.0 = intent;
     }
```
Comment (MiniaczQ) [inline src/game/movement.rs:44]: I'd go as far as `get_single_mut` here

File: src/game/movement.rs [hunk 2]
```diff
@@ -13,32 +13,31 @@ pub(super) fn plugin(app: &mut App) {
...
-            velocity.x -= 1.0;
-        }
-        if keyboard_input.pressed(KeyCode::KeyD) {
-            velocity.x += 1.0;
-        }
-
-        // Need to normalize and scale because otherwise
-        // diagonal movement would be faster than horizontal or vertical movement.
-        velocity.0 = velocity.normalize_or_zero() * SPEED;
+    let mut intent = Vec3::ZERO;
```
Comment (janhohenheim) [inline src/game/movement.rs:25]: Not a fan of this rename. @MiniaczQ, opinion?

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: src/game/movement.rs (hunk 2 above)
Comment (benfrankel) [reply src/game/movement.rs:25]: Rationale: Input -> Intent -> Behavior opens the door for EnemyAi -> Intent -> Behavior, so this is the name I would use for the intermediate value.

Comment (benfrankel) [reply src/game/movement.rs:25]: Although I suppose the intent wouldn't have speed applied.

Comment (MiniaczQ) [reply src/game/movement.rs:25]: I'm ok with `intent` as long as it's axis-like, after we multiply it with SPEED it's very clearly `velocity` and no longer `intent`.

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**